client/watcher/watcher.go

```diff
@@ -71,29 +72,43 @@ func (w *Watcher) Init() error {
 func (w *Watcher) Watch() {
 	done := make(chan bool)
 	go func() {
+		defer close(done)
+
 		for {
 			w.mu.Lock()
 			select {
-			case event := <-w.watcher.Events:
-				if event.Op&fsnotify.Write == fsnotify.Write ||
-					event.Op&fsnotify.Create == fsnotify.Create ||
-					event.Op&fsnotify.Chmod == fsnotify.Chmod {
+			case event, ok := <-w.watcher.Events:
+				if !ok {
+					w.mu.Unlock()
+					return
+				}
+
+				correctFile := event.Name != "" && strings.Contains(w.path, path.Clean(event.Name))
+				isWriteEvent := (event.Op&fsnotify.Write == fsnotify.Write) || (event.Op&fsnotify.Create == fsnotify.Create) (event.Op&fsnotify.Rename == fsnotify.Rename)
+
+				printer.Say("received fsnotify event: %v %v. Path: %v Correct file: %v, Write: %v", event.Op, event.Name, w.path, correctFile, isWriteEvent)
+
+				if correctFile && isWriteEvent {
+					printer.Say("handling fsnotify event: %v %v", event.Op, event.Name)
 					err := w.UpdateBytes()
 					if err != nil {
-						printer.LogErrf("UpdateBytes error: %v", err)
+						printer.Err("UpdateBytes error: %v", err)
 					}
 
 					// Rewatch the path
-					err = w.watcher.Add(w.path)
+					err = w.watcher.Add(path.Dir(w.path))
```

Comment:
do we need to rewatch this, since we are now watching the parent directory?
though the `fsnotify` code looks like it does no harm if we watch again...

Reply:
If the directory that we're watching itself changes, we'll need to rewatch